base/shell/explorer/taskswnd.cpp

```diff
@@ -1858,6 +1858,13 @@ class CTaskSwitchWnd :
             UpdateButtonsSize(FALSE);
         }
 
+        if (newSettings->bSmallIcons != g_TaskbarSettings.bSmallIcons)
+        {
+            g_TaskbarSettings.bSmallIcons = newSettings->bSmallIcons;
+            RefreshWindowList();
+            UpdateButtonsSize(FALSE);
```

Comment:
Note that in case both the bGroupButtons and the bSmallIcons are changed, you are doing twice the RefreshWindowList and the UpdateButtonsSize . This is of course redundant and needs to be fixed.

For example in that function, by having a BOOL bSettingsChanged that gets set to TRUE whenever either of those settings have been changed, and in case it's TRUE, do those function calls only once.

Reply:
Added `bSettingsChanged` Boolean and fixed the double refresh bug.